tools/AzDev/src/Services/DefaultCodebaseProvider.cs

```diff
@@ -12,34 +12,47 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
+using System;
 using System.IO.Abstractions;
 using AzDev.Models.Inventory;
 
 namespace AzDev.Services
 {
     internal class DefaultCodebaseProvider : ICodebaseProvider
     {
-        private IContextProvider _contextProvider;
-        private IFileSystem _fs;
+        private ILogger _logger;
```

Comment:
default logger or any safeguard to guarantee _logger was initialized before being used?

Reply:
Makes sense. Wouldn't want to blocker the main logic if logger is not ready. I'll assign it a default value.
Note that this is not thread safe